src/Microsoft.FeatureManagement/Targeting/ContextualTargetingFeatureVariantAssigner.cs

```diff
@@ -67,69 +67,128 @@ public ValueTask<FeatureVariant> AssignVariantAsync(FeatureVariantAssignmentCont
 
             FeatureVariant variant = null;
 
-            double cumulativePercentage = 0;
+            var lookup = new Dictionary<FeatureVariant, TargetingFilterSettings>();
+
+            //
+            // Check users
+            foreach (FeatureVariant v in featureDefinition.Variants)
+            {
+                TargetingFilterSettings targetingSettings = v.AssignmentParameters.Get<TargetingFilterSettings>();
+
+                //
+                // Put in lookup table to avoid repeatedly creating targeting settings
+                lookup[v] = targetingSettings;
+
+                if (targetingSettings == null &&
+                    v.Default)
```

Comment:
Should this condition be OR instead of AND?
What is the purpose of specifying any audience for default variant?

Reply:
It is intentionally AND. 

Here's an example setup with default variant that has an audience.

Option A is the default, Option B targets 50% of users by default rollout percentage.
Since A is the default, this setup means A will target 50% of the user base by default.
But we specifically want "Jeff" to use option A. So we setup an audience for option A and target Jeff.